src/k8s-engine/homestead.js

```diff
@@ -71,23 +71,69 @@ async function ensureHelmChart(state) {
   return false;
 }
 
+/**
+ * Ensure that we can actually make a connection; sometimes the first few get
+ * lost, possibly because the underlying server is not ready yet.
+ * @param {string} namespace The namespace to forward to.
+ * @param {string} endpoint The endpoint in the namespace to forward to.
+ * @param {number} port The port to forward to on the endpoint.
+ * @param {KubeClient} client
+ */
+async function waitForConnection(namespace, endpoint, port, client) {
+  const targetName = `${namespace}/${endpoint}:${port}`;
+  for (; ;) {
+    const listeningPort = await client.getForwardedPort(namespace, endpoint, port);
+    if (!listeningPort) {
+      await new Promise(resolve => setTimeout(resolve, 1000));
+      continue;
+    }
+    try {
+      await new Promise((resolve, reject) => {
+        console.log(`Attempting to make probe request for ${targetName}...`);
+        const req = https.get({ port: listeningPort, rejectUnauthorized: false }, response => {
+          response.destroy();
+          if (response.statusCode >= 200 && response.statusCode < 400) {
+            return resolve();
+          }
+          reject(`Got unexpected response ${response?.statusCode}`);
+        });
+        req.on('close', reject);
+        req.on('error', reject);
+        // Manually reject on a time out
```

Comment:
So if we get to this code (because we didn't resolve at line 96) we're going to reject, but why the 5-second delay?

Reply:
Because the `https.get()` can take some time to happen.  We want to give it _some_ time before retrying, after all; everything here runs concurrently.